src/Agent.Plugins/PipelineCache/SavePipelineCacheV0.cs

```diff
@@ -63,6 +63,14 @@ public override async Task RunAsync(AgentTaskPluginExecutionContext context, Can
             CancellationToken token)
         {
             string contentFormatValue  = context.Variables.GetValueOrDefault(ContentFormatVariableName)?.Value ?? string.Empty;
+            string calculatedFingerPrint = context.TaskVariables.GetValueOrDefault(ResolvedFingerPrintVariableName)?.Value ?? string.Empty;
+
+            if(!string.IsNullOrWhiteSpace(calculatedFingerPrint) && !fingerprint.ToString().Equals(calculatedFingerPrint, StringComparison.Ordinal))
```

Comment:
>Ordinal [](start = 139, length = 7)

casing doesn't matter I assume

Reply:
Yup. Discussed offline. Since we need case-sensitivity, we are using the Ordinal comparer.